apps/openmw/mwclass/npc.cpp

```diff
@@ -58,116 +62,47 @@ namespace
         return new NpcCustomData (*this);
     }
 
-    int is_even(double d) {
-        double int_part;
-        modf(d / 2.0, &int_part);
-        return 2.0 * int_part == d;
-    }
-
-    int round_ieee_754(double d) {
-        double i = floor(d);
-        d -= i;
-        if(d < 0.5)
-            return static_cast<int>(i);
-        if(d > 0.5)
-            return static_cast<int>(i) + 1;
-        if(is_even(i))
-            return static_cast<int>(i);
-        return static_cast<int>(i) + 1;
-    }
-
-    void autoCalculateAttributes (const ESM::NPC* npc, MWMechanics::CreatureStats& creatureStats)
+    class Stats : public GamePlay::StatsBase
     {
-        // race bonus
-        const ESM::Race *race =
-            MWBase::Environment::get().getWorld()->getStore().get<ESM::Race>().find(npc->mRace);
+        MWMechanics::CreatureStats& mCreatureStats;
+        MWMechanics::NpcStats& mNpcStats;
 
-        bool male = (npc->mFlags & ESM::NPC::Female) == 0;
+    public:
 
-        int level = creatureStats.getLevel();
-        for (int i=0; i<ESM::Attribute::Length; ++i)
-        {
-            const ESM::Race::MaleFemale& attribute = race->mData.mAttributeValues[i];
-            creatureStats.setAttribute(i, male ? attribute.mMale : attribute.mFemale);
-        }
+        Stats(MWMechanics::CreatureStats& creatureStats, MWMechanics::NpcStats& npcStats)
```

Comment:
You don't need both CreatureStats and NpcStats. NpcStats is derived from CreatureStats (although it wasn't at the time the original autocalc was written).